flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/parser/ClientParser.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.client.cli.parser;
+
+import org.apache.flink.sql.parser.impl.FlinkSqlParserImplTokenManager;
+import org.apache.flink.sql.parser.impl.SimpleCharStream;
+import org.apache.flink.sql.parser.impl.Token;
+import org.apache.flink.table.client.gateway.SqlExecutionException;
+import org.apache.flink.table.operations.Operation;
+
+import javax.annotation.Nonnull;
+
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/** ClientParser use {@link FlinkSqlParserImplTokenManager} to do lexical analysis. */
+public class ClientParser implements SqlCommandParser {
+
+    /** A dumb implementation. TODO: remove this after unifying the SqlMultiLineParser. */
+    @Override
+    public Optional<Operation> parseCommand(String command) {
+        return Optional.empty();
+    }
+
+    public Optional<StatementType> parseStatement(@Nonnull String statement)
+            throws SqlExecutionException {
+        String trimmedStatement = statement.trim();
+        FlinkSqlParserImplTokenManager tokenManager =
+                new FlinkSqlParserImplTokenManager(
+                        new SimpleCharStream(new StringReader(trimmedStatement)));
+        List<Token> tokenList = new ArrayList<>();
+        Token token;
+        do {
+            token = tokenManager.getNextToken();
+            tokenList.add(token);
+        } while (token.endColumn != trimmedStatement.length());
```

Comment:
I think we can condition can be `token != null`

Reply:
After checking the codes of `getNextToken`, I found that when the input stream is over, it won't return `null` but a `EOF` token. So I think  use condition `token.kind != EOF` may be better?